### ISSUE TITLE
 DbtCloudRunJob task fails with nested input for additional_args

### ISSUE THREAD

**Issue by olivito** [PrefectHQ/prefect#5706]
## Description

If I call the `DbtCloudRunJob` with nested inputs for the optional parameter `additional_args`, the DBT API returns a 400 response "Bad Request":
```
Task 'DbtCloudRunJob': Exception encountered during task execution!
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/prefect/engine/task_runner.py", line 876, in get_task_run_state
    value = prefect.utilities.executors.run_task_with_timeout(
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/executors.py", line 467, in run_task_with_timeout
    return task.run(*args, **kwargs)  # type: ignore
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/tasks.py", line 456, in method
    return run_method(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/prefect/tasks/dbt/dbt.py", line 401, in run
    run = trigger_job_run(
  File "/usr/local/lib/python3.9/dist-packages/prefect/tasks/dbt/dbt_cloud_utils.py", line 110, in trigger_job_run
    raise TriggerDbtCloudRunFailed(trigger_request.reason)
prefect.tasks.dbt.dbt_cloud_utils.TriggerDbtCloudRunFailed: Bad Request
```

## Expected Behavior

I expect to get a 200 response from the DBT API and for the job to start correctly with the override properties from `additional_args`.

## Reproduction

The one nested parameter supported by the [DBT Trigger Job Run API](https://docs.getdbt.com/dbt-cloud/api-v2#operation/triggerRun) is `steps_override` (Array of strings).  When I include that, I get the 400 response described above:
```
from prefect import Flow
from prefect.tasks.dbt.dbt import DbtCloudRunJob

dbt_cloud_run_job = DbtCloudRunJob()

# set DBT_CLOUD_TOKEN environment variable
with Flow("dbt_override_steps_minimal_flow") as flow:
    result = dbt_cloud_run_job(
        cause="triggered by Prefect to test steps override",
        account_id=12345,
        job_id=1234,
        wait_for_job_run_completion=True,
        additional_args={
            "schema_override": "my_schema",
            "steps_override": [
                "dbt run --select +my_model",
                "dbt test --select +my_model",
            ],
        },
    )

flow.run()
```

If I comment out `steps_override` from `additional_args`, then the job is submitted successfully and runs.

From testing locally, I believe the issue is this line:
https://github.com/PrefectHQ/prefect/blob/4462f2300cdc5da765d914030db1ef82e697d8e8/src/prefect/tasks/dbt/dbt_cloud_utils.py#L107

instead of `data=data` (which assumes a string), it should be `json=data` (which assumes a json dictionary). 
I'm able to call the DBT API directly using the python requests library with that change.

I can put up a PR for this one line fix, though I don't have any particular ideas on how to test for this in the unit tests.

## Environment

I am running with prefect 0.15.13, but I see the same issue in the source code for 1.2.0.
```
{
  "config_overrides": {},
  "env_vars": [],
  "system_information": {
    "platform": "macOS-12.3.1-x86_64-i386-64bit",
    "prefect_backend": "cloud",
    "prefect_version": "0.15.13",
    "python_version": "3.8.9"
  }
}
```

**Comment by tirkarthi** [comment]
I guess this can be closed with https://github.com/PrefectHQ/prefect/pull/5709 merged

**Comment by olivito** [comment]
Yes, this is resolved by #5709.
